File: 백준/Silver/10828. 스택/스택.py
import sys

user_input = int(input())
stack = []
for _ in range(user_input):
    input_data = sys.stdin.readline().strip().split(" ")
    if input_data[0] == "push":
        stack.append(int(input_data[1]))
    elif input_data[0] == "pop" and len(stack)>=1:
        print(stack.pop())
    elif input_data[0] == "pop" and len(stack)==0:
        print(-1)
    elif input_data[0] == "size":
        print(len(stack))
    elif input_data[0] == "empty" and len(stack)==0:
        print(1)
    elif input_data[0] == "empty" and len(stack)>=1:
        print(0)
    elif input_data[0] == "top" and stack != [] :
        print(stack[-1])
    else:
        print(-1)

File: 백준/Silver/10828. 스택/스택.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
const stack = [];

const result = [];

rl.on('line', function (line) {
  input.push(line);
  if (input.length > 1) {
    const [command, x] = line.trim().split(' ');
    if (command === 'push') stack.push(x);
    else if (command === 'pop') {
      stack.length ? result.push(+stack.pop()) : result.push(-1);
    } else if (command === 'size') {
      result.push(stack.length);
    } else if (command === 'empty') {
      stack.length ? result.push(0) : result.push(1);
    } else if (command === 'top') {
      stack.length ? result.push(stack[stack.length - 1]) : result.push(-1);
    } else {
      console.error('non command');
    }
  }

  if (+input[0] === input.length - 1) {
    console.log(result.join('\n'));
    rl.close();
  }
}).on('close', function () {
  process.exit(); // 프로세스 종료
});
